File: AndroidStudioStudy/3-AndroidJetpackApplicationGuide/Chapter2-LifeCycle/App1-LifeCycle/app/src/main/java/com/bo/app1_lifecycle/activity/MainActivity.java
package com.bo.app1_lifecycle.activity;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.bo.app1_lifecycle.Listener.MyLocationListener;
import com.bo.app1_lifecycle.R;

public class MainActivity extends AppCompatActivity {

    private MyLocationListener myLocationListener;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        myLocationListener=new MyLocationListener(this, new MyLocationListener.OnLocationChangedListener() {
            @Override
            public void onChanged(double latitude, double longitude) {
                //todo 展示收到的位置信息
            }
        });

        //将观察者与被观察者绑定
        getLifecycle().addObserver(myLocationListener);
    }

    @Override
    protected void onResume() {
        super.onResume();

        /*//开始获取用户的地理位置
        startGetLocation();*/
    }

    @Override
    protected void onPause() {
        super.onPause();

        /*//停止获取用户的地理位置
        stopGetLocation();*/
    }

}

File: Internship-Android-Markdown/IPC/Binder/client/src/main/java/com/xiaoyingbo/binder/MainActivity.kt
package com.xiaoyingbo.binder

import android.content.ComponentName
import android.content.Context
import android.content.Intent
import android.content.ServiceConnection
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.IBinder
import android.util.Log
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    private lateinit var iTestBinder:ITestInterface

    private val connect:ServiceConnection=object : ServiceConnection{
        override fun onServiceConnected(name: ComponentName?, service: IBinder?) {
            Log.e("aidl","onServiceConnected success");
            iTestBinder=ITestInterface.Stub.asInterface(service);
        }

        override fun onServiceDisconnected(name: ComponentName?) {
            //TODO("Not yet implemented")
        }

    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val intent = Intent()
        /*intent.setPackage("com.xiaoyingbo.server");
        intent.setClassName("com.xiaoyingbo.server","AIDLService")*/
        intent.component = ComponentName("com.xiaoyingbo.server","com.xiaoyingbo.server.AIDLService")
        bindService(intent,connect,Context.BIND_AUTO_CREATE)

        findViewById<TextView>(R.id.tv).setOnClickListener{
            iTestBinder.test()
        }
    }

    override fun onDestroy() {
        super.onDestroy()

        unbindService(connect)
    }
}